### PR TITLE
Added dialogue, QuestSystem and DialogueSystem

### PR DESCRIPTION
**Pull request by accidentallyc**
* Only 1 Dialogue can only be activated at a time
* It turns off when you walk away
* Dialogues can fire off `set_quest` after its done
* Dialogues can dynamically read off the quest dictionary via `quest_var`

image

image

### CODE REVIEW

File: Components/dialogue/GameDialogue.gd [hunk 1]
```diff
@@ -0,0 +1,41 @@
+extends Resource
+
+class_name GameDialogue
+
+## Example
+##  get_quest('plants_cut') >= 4 AND get_quest('meteor') == 'done'
+@export_multiline var condition:String
```
Comment (accidentallyc) [inline Components/dialogue/GameDialogue.gd:7]: Any godot expression. Special func `get_quest` to access `QuestSystem.quest_vars`

File: Entities/Player/player.gd [hunk 2]
```diff
@@ -17,75 +17,75 @@ enum FacingDirection {UP, RIGHT, DOWN, LEFT}
 
 
 func _ready() -> void:
```
Comment (accidentallyc) [inline Entities/Player/player.gd:19]: Uh i didnt change anything here - the damn autospacer thing

Comment (accidentallyc) [reply Entities/Player/player.gd:19]: image
Do we have different settings?

Comment (JJPSeno) [reply Entities/Player/player.gd:19]: Oh yeah we do haha. I'll change mine to match yours.
image

**Pull request merged**